core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -486,7 +492,7 @@ class KafkaController(val config: KafkaConfig, zkClient: KafkaZkClient, time: Ti
     info(s"New partition creation callback for ${newPartitions.mkString(",")}")
     partitionStateMachine.handleStateChanges(newPartitions.toSeq, NewPartition)
     replicaStateMachine.handleStateChanges(controllerContext.replicasForPartition(newPartitions).toSeq, NewReplica)
-    partitionStateMachine.handleStateChanges(newPartitions.toSeq, OnlinePartition, Option(OfflinePartitionLeaderElectionStrategy))
+    partitionStateMachine.handleStateChanges(newPartitions.toSeq, OnlinePartition, Some(OfflinePartitionLeaderElectionStrategy))
```

Comment:
It is fine here but the nice thing of using `Option` or more precisely `Option.apply` is that it handles `null` and returns the type `Option[T]` instead of `Some[T]` by `Some.apply`.

Reply:
Yeah, I just changed to `Some` because we have a non-nullable object. I thought it is a little cleaner, but it's a bit subjective.